scripts/Engineering/EnginX.py

```diff
@@ -381,17 +582,22 @@ def create_difc_zero_workspace(difc, tzero, crop_on, name):
         actual_i = correction + i
         # retrieve required workspace
         if not plot_spec_num:
-            bank_ws = simple.AnalysisDataService.retrieve("engg_calibration_bank_{}".format(actual_i))
+            if actual_i == 1:
+                fitparam_ws = simple.AnalysisDataService.retrieve("diag_North_fitparam")
+            elif actual_i == 2:
+                fitparam_ws = simple.AnalysisDataService.retrieve("diag_South_fitparam")
         else:
-            bank_ws = simple.AnalysisDataService.retrieve(name)
+            fitparam_ws = simple.AnalysisDataService.retrieve("diag_Cropped_fitparam")
+
+        expected_dspacing_peaks = Utils.default_ceria_expected_peaks(final=True)
 
         # get the data to be used
         x_val = []
         y_val = []
         y2_val = []
-        for irow in range(0, bank_ws.rowCount()):
-            x_val.append(bank_ws.cell(irow, 0))
-            y_val.append(bank_ws.cell(irow, 5))
+        for irow in range(0, fitparam_ws.rowCount()):
+            x_val.append(expected_dspacing_peaks[-(irow+1)])
+            y_val.append(fitparam_ws.cell(irow, 5))
             y2_val.append(x_val[irow] * difc[i - 1] + tzero[i - 1])
```

Comment:
1. Why is there no `difa` here and what exactly is `y2` - is it meant to be the fitted quadratic curve to TOF(d)? If so perhaps we call this `yfit`? Generally speaking I don't like the variables names `x_val`, `y_val `etc. (I know they pre-date your changes but since you're familiar with their purpose it would make sense for you to change them in this PR).
2. I think there is a unit converter exposed to python that will convert single values (not workspaces) based of the diffractometer constants so we don't have to reproduce this equation each time. I'm having trouble finding it, @DannyHindson do you know?
3. Also it would be good to retrieve the errors on the TOF as well to plot.

Reply:
Ah here I think is how to use the UnitConverter in python
https://github.com/mantidproject/mantid/blob/aca78985607ac37e8a57ef3f16774e891b9b106d/scripts/Engineering/gui/engineering_diffraction/tabs/fitting/data_handling/data_model.py#L340